app/main.py

```diff
@@ -1,2 +1,123 @@
+import dataclasses
+from typing import Hashable
+
+
+@dataclasses.dataclass
+class Node:
+    key: object
+    key_hash: int
+    value: object
+
+
 class Dictionary:
-    pass
+    __load_factor = 2 / 3
+
+    def __init__(self) -> None:
+        self.__hashtable = [None] * 8
+
+    def __setitem__(self, key: Hashable, value: object) -> None:
+        hashed_key = hash(key)
+        node = Node(key, hashed_key, value)
+        self.__choose_cell(node)
+        self.__check_overload()
+
+    def __getitem__(self, key: Hashable) -> object:
+        hashed_key = hash(key)
+        length = len(self.__hashtable)
+        preferred_cell = hashed_key % length
+        for i in range(preferred_cell, length):
+            if self.__hashtable[i] and self.__hashtable[i].key == key:
+                return self.__hashtable[i].value
+        for i in range(0, preferred_cell):
+            if self.__hashtable[i] and self.__hashtable[i].key == key:
+                return self.__hashtable[i].value
+        raise KeyError("Key not found")
+
+    def __len__(self) -> int:
+        return len([node for node in self.__hashtable if node is not None])
+
+    def clear(self) -> None:
+        self.__hashtable = [None] * 8
+
+    def __delitem__(self, key: Hashable) -> None:
+        hashed_key = hash(key)
+        length = len(self.__hashtable)
+        preferred_cell = hashed_key % length
+        for i in range(preferred_cell, length):
+            if self.__hashtable[i] and self.__hashtable[i].key == key:
+                self.__hashtable[i] = None
+                return
+        for i in range(0, preferred_cell):
+            if self.__hashtable[i] and self.__hashtable[i].key == key:
+                self.__hashtable[i] = None
+                return
+        raise KeyError("Key not found")
+
+    def get(self, key: object) -> object:
+        try:
+            return self.__getitem__(key)
+        except KeyError:
+            return -1
+
+    def update(self, second_dict: "Dictionary") -> None:
+        for node in second_dict.__dict__["_Dictionary__hashtable"]:
+            if node:
+                self.__setitem__(node.key, node.value)
+            self.__setitem__(node.key, node.value)
+
+    def __iter__(self) -> list:
+        self.index = 0
+        return self.__hashtable
```

Comment:
`dict` when being iterated sends us the keys, if not values or items methods were specified

Reply:
Understood, accepted.